src/overlays/actors/ovl_Boss_Va/z_boss_va.c

```diff
@@ -173,157 +319,3695 @@ static ColliderQuadInit D_8095BFF4 = {
+                Matrix_MultVec3f(&sp70, &this->effectPos[6]);
+                sp70.z -= 33.0f;
+                if (sp70.z < 0.0f) {
+                    sp70.z = 0.0f;
+                }
+            } else {
+                sp70.z -= 22.0f;
+                if (sp70.z < 0.0f) {
+                    sp70.z = 0.0f;
+                }
+                Matrix_MultVec3f(&sp70, &this->effectPos[6]);
+                sp70.z -= 22.0f;
+                if (sp70.z < 0.0f) {
+                    sp70.z = 0.0f;
+                }
+            }
+            Matrix_MultVec3f(&sp70, &this->effectPos[5]);
+            Matrix_MultVec3f(&sp64, &this->colliderLightning.dim.quad[1]);
+            Matrix_MultVec3f(&sp58, &this->colliderLightning.dim.quad[0]);
+            Matrix_MultVec3f(&sp4C, &this->colliderLightning.dim.quad[3]);
+            Matrix_MultVec3f(&sp40, &this->colliderLightning.dim.quad[2]);
+            Collider_SetQuadVertices(&this->colliderLightning, &this->colliderLightning.dim.quad[0],
+                                     &this->colliderLightning.dim.quad[1], &this->colliderLightning.dim.quad[2],
+                                     &this->colliderLightning.dim.quad[3]);
+            Matrix_Pull();
+            break;
+    }
+}
+
+s32 BossVa_BariOverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                                void* thisx) {
+    BossVa* this = THIS;
+
+    switch (limbIndex) {
+        case 2:
+            *dList = NULL;
+            break;
+        case 3:
+            Matrix_Scale(this->unk_1D8.x, 1.0f, this->unk_1D8.x, MTXMODE_APPLY);
+            break;
+        case 4:
+            Matrix_Scale(1.0f, this->unk_1D8.y, 1.0f, MTXMODE_APPLY);
+            break;
+    }
+    return 0;
+}
+
+void BossVa_BariPostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, void* thisx) {
+    BossVa* this = THIS;
+
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_boss_va.c", 4494);
+
+    if (limbIndex == 2) {
+        gSPSegment(POLY_XLU_DISP++, 0x0A,
+                   Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0, (globalCtx->gameplayFrames * 10) % 32, 0x10, 0x20, 1,
+                                    0, (globalCtx->gameplayFrames * -5) % 32, 0x10, 0x20));
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_va.c", 4508),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, gBarinadeDL_000FA0);
+    } else if ((limbIndex == 3) || (limbIndex == 4)) {
+        gSPMatrix(POLY_XLU_DISP++, Matrix_NewMtx(globalCtx->state.gfxCtx, "../z_boss_va.c", 4512),
+                  G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+        gSPDisplayList(POLY_XLU_DISP++, *dList);
+    }
+
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_boss_va.c", 4517);
+}
+
+void BossVa_Draw(Actor* thisx, GlobalContext* globalCtx) {
+    s16* fakeParams; // Yeah this is the fakest shit ever. I don't know.
```

Comment:
I wouldn't commit inappropriate language to the repo ngl.

Also idk why `fakeParams`, imo `paramsPtr` would be more appropriate

Reply:
To emphasize its fakeness. That stack slot is almost certainly actually globalCtx2, but this is the only match I could find. Hopefully someone can find a better one.

Also, yeah, that comment will change. I was just kind of frustrated with this function...>_>